### PR TITLE
add unit tests for python cell_census package

### PR DESCRIPTION
**Pull request by bkmartinjr**
Changes - more/better unit tests for the cell_census python package:
* add error condition unit test coverage for `open_soma` and other API
* add tests to cover all `open_soma` parameters
* add missing absolute URI test for `uri_join`

### CODE REVIEW

File: api/python/cell_census/tests/test_open.py [hunk 1]
```diff
@@ -37,16 +71,42 @@ def test_get_source_h5ad_uri() -> None:
...
         census_datasets = census["census_info"]["datasets"].read().concat().to_pandas()
 
     small_dataset = census_datasets.nsmallest(1, "dataset_total_cell_count").iloc[0]
+    assert isinstance(small_dataset.dataset_id, str)
+    return small_dataset.dataset_id
+
+
+@pytest.mark.live_corpus
+def test_download_source_h5ad(tmp_path: pathlib.Path, small_dataset_id: str) -> None:
+    # with cell_census.open_soma(census_version="latest") as census:
```
Comment (ebezzi) [inline api/python/cell_census/tests/test_open.py:91]: Can the commented code be removed?

File: api/python/cell_census/tests/test_directory.py [hunk 2]
```diff
@@ -60,6 +63,11 @@ def test_get_census_version_directory(directory_mock: Any) -> None:
         assert directory[tag] == cell_census.get_census_version_description(tag)
 
 
+def test_get_census_version_description_errors() -> None:
+    with pytest.raises(KeyError):
```
Comment (ebezzi) [inline api/python/cell_census/tests/test_directory.py:67]: Not strictly related to the tests, but I was wondering if maybe we don't want to specialize these exceptions? Right now everything is `KeyError` or `ValueError` but these exceptions map to very specific issues (for instance, non existent census version.

**Review by ebezzi — APPROVED** [review]
LGTM, one nit and one more generic comment (not strictly related to this PR).

File: api/python/cell_census/tests/test_open.py (hunk 1 above)
Comment (bkmartinjr) [reply api/python/cell_census/tests/test_open.py:91]: oops!  Yes, definitely.

File: api/python/cell_census/tests/test_directory.py (hunk 2 above)
Comment (bkmartinjr) [reply api/python/cell_census/tests/test_directory.py:67]: sure, we could do that.

I agree it is a separate issue and I'll file it.

**Pull request merged**